### PR TITLE
feat: bump rollkit version

### PR DESCRIPTION
**Pull request by MSevey**
## Overview

This pulls in the breaking change of rollkit v0.14.x sequencing API

## Summary by CodeRabbit

- **New Features**
	- Updated documentation for integrating RappDK with cometBFT, including new versioning information and migration notices.
	- Introduced a new `x/sequencer` module for managing sequencer functionality.

- **Documentation**
	- Streamlined links in the Rollkit SDK README for easier navigation.
	- Enhanced clarity in the default app template documentation regarding module removal and app structure changes.

- **Chores**
	- Comprehensive updates to various dependencies to ensure compatibility and incorporate improvements.

### CODE REVIEW

**Comment by MSevey** [comment]
@coderabbitai ignore

**Comment by MSevey** [comment]
@coderabbitai ignore

File: go.mod [hunk 1]
```diff
@@ -203,23 +198,23 @@ require (
...
 	github.com/quic-go/qpack v0.4.0 // indirect
 	github.com/quic-go/quic-go v0.46.0 // indirect
 	github.com/quic-go/webtransport-go v0.8.0 // indirect
 	github.com/raulk/go-watchdog v1.3.0 // indirect
 	github.com/rcrowley/go-metrics v0.0.0-20201227073835-cf1acfcdf475 // indirect
 	github.com/rogpeppe/go-internal v1.12.0 // indirect
-	github.com/rollkit/go-da v0.8.0 // indirect
-	github.com/rollkit/go-sequencing v0.2.1-0.20241010053131-3134457dc4e5 // indirect
+	github.com/rollkit/go-da v0.9.0 // indirect
+	github.com/rollkit/go-sequencing v0.4.0 // indirect
```
Comment (yarikbratashchuk) [inline go.mod:212]: why not 0.4.1

**Review by yarikbratashchuk — DISMISSED** [review]

File: go.mod (hunk 1 above)
Comment (MSevey) [reply go.mod:212]: wasn't released when the PR was created. I can bump.

**Review by Manav-Aggarwal — APPROVED** [review]

**Pull request merged**